src/garage/tf/models/gaussian_conv_model.py

```diff
@@ -0,0 +1,301 @@
+        std_filter_dims(tuple[int]): Dimension of the filters. For example,
+            (3, 5) means there are two convolutional layers. The filter
+            for first layer is of dimension (3 x 3) and the second one is of
+            dimension (5 x 5).
+        std_num_filters(tuple[int]): Number of filters. For example, (3, 32)
+            means there are two convolutional layers. The filter for the first
+            layer has 3 channels and the second one with 32 channels.
+        std_strides(tuple[int]): The stride of the sliding window. For example,
+            (1, 2) means there are two convolutional layers. The stride of the
+            filter for first layer is 1 and that of the second layer is 2.
+        std_padding (str): The type of padding algorithm to use in std network,
+            either 'SAME' or 'VALID'.
+        std_hidden_sizes (list[int]): Output dimension of dense layer(s) for
+            the Conv for std. For example, (32, 32) means the Conv consists
+            of two hidden layers, each with 32 hidden units.
+        min_std (float): If not None, the std is at least the value of min_std,
+            to avoid numerical issues.
+        max_std (float): If not None, the std is at most the value of max_std,
+            to avoid numerical issues.
+        std_hidden_nonlinearity: Nonlinearity for each hidden layer in
+            the std network.
+        std_output_nonlinearity (callable): Activation function for output
+            dense layer in the std network. It should return a tf.Tensor. Set
+            it to None to maintain a linear activation.
+        std_output_w_init (callable): Initializer function for the weight
+            of output dense layer(s) in the std network.
+        std_parametrization (str): How the std should be parametrized. There
+            are two options:
+            - exp: the logarithm of the std will be stored, and applied a
+               exponential transformation
+            - softplus: the std will be computed as log(1+exp(x))
+        layer_normalization (bool): Bool for using layer normalization or not.
+    """
+
+    def __init__(self,
+                 output_dim,
+                 filter_dims,
+                 num_filters,
+                 strides,
+                 padding,
+                 hidden_sizes,
+                 name=None,
+                 hidden_nonlinearity=tf.nn.tanh,
+                 hidden_w_init=tf.glorot_uniform_initializer(),
+                 hidden_b_init=tf.zeros_initializer(),
+                 output_nonlinearity=None,
+                 output_w_init=tf.glorot_uniform_initializer(),
+                 output_b_init=tf.zeros_initializer(),
+                 learn_std=True,
+                 adaptive_std=False,
+                 std_share_network=False,
+                 init_std=1.0,
+                 min_std=1e-6,
+                 max_std=None,
+                 std_filter_dims=[],
+                 std_num_filters=[],
```

Comment:
doesn't filter_dims tell me how many filters there should be?

Reply:
`filter_dims` specifies the dimension of the filters, `num_filters` specifies the number of filters to use of that dimension, `std_*` arguments specify the same parameters for the `std` CNN.

Reply:
are you saying that std_filter_dims and std_num_filters is a run-length encoding? i.e. the total number of filters is `sum(std_num_filters)`?

Reply:
yes, i guess so. `num_filters = (3, 6)` and `filter_dims = (5, 3)` implies 3 filters of  shape `(5x5)` and 6 filters of shape `(3x3)`

Reply:
ok that is a strange way of doing it and hard to test/validate. can you file an issue to change the API to just use a list of lists/tuple of tuples?

it can be the same issue as the comment above about filter_dims

Reply:
Sure, I'll file one now